tests/testthat/test-names.R

```diff
@@ -8,6 +8,8 @@ test_that("vec_names() retrieves names", {
   expect_identical(vec_names(set_names(letters)), letters)
   expect_null(vec_names(mtcars))
   expect_identical(vec_names(Titanic), dimnames(Titanic)[[1]])
+  x <- matrix(1L, dimnames = list("row", "col"))
+  expect_identical(vec_names(x), dimnames(x)[[1]])
```

Comment:
Perhaps also a test with multiple rows?

Reply:
The Titanic example is a 3D table with multiple rows, so I thought that sufficed. It happened to pass with the previous implementation of `vec_dim()`. This length 1 case was just to expose the issue.